### ISSUE TITLE
As a CIF administrator I want to be able to automatically populate the Report Due Date after entering the Substantial Deadline so that I don't have to manually set or track the date

### ISSUE THREAD

**Issue by suhafa** [bcgov/cas-cif#593]
#### Description:

Milestone reports have a date by when the work should be finished by. It can be automatically calculated based upon the type of milestone. The user wants to generate the Report Due Date automatically after entering the Substantial Completion Deadline, and vice versa; setting the Report Due Date sets the Substantial Completion Date to 30 days prior. The Report Due Date is always 30 days after Substantial Completion Deadline for General, Advanced and Reporting Milestones.
(Note: For Project Summary Report milestone it is due 60 days after Substantial Completion Deadline and 150 days after Substantial Completion Deadline for TEIMP report - but these are no longer Milestone types we will have on the Milestone Reports page. They are now separate pages of their own)

Wireframe:
![image.png](https://images.zenhubusercontent.com/6201b10c9827341d8d5cbe9c/0ea51fb3-4778-41b6-a66a-48e45385f912)
or see here in [Figma](https://www.figma.com/file/PzOY8RyQnVXnjZJdSv2AqQ/CIF-Design?node-id=1209%3A2663&t=wraNoTh57o6QJRyk-1)

#### Acceptance Criteria:
*Note* AC has been updated as the intended behaviour of this changed during development. Comments below.

Given that a user is creating a milestone
And they’ve selected milestone type as General, Advanced or Reporting
When the Substantial Completion Deadline is set
Then the date for Report Due Date will be set to 30 days after the substantial completion

Given that a user is creating a milestone
And they’ve selected milestone type as General, Advanced or Reporting
When the Report Due Date is set
Then the date for Substantial Completion Deadline will be set to 30 days prior to the  Report Due Date

**No Longer Relevant**
Given that a user is creating a milestone
And no Substantial Completion Deadline is set
And no milestone type is selected
Then the Report Due Date field will be blank

Given that a user is creating a milestone
And they’ve selected milestone type as General, Advanced or Reporting
And no Substantial Completion Deadline is set
Then an error will appear below Report Due Date which states, "Please enter the Substantial Completion Deadline first” 
So that the user doesn't try to set the deadline manually before it's automatically calculated

#### Development Checklist:

- [x] The dates are correctly calculated and update automatically on the front end when updating the milestone type
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [x] User story is included
- [x] User role and type are identified
- [x] Acceptance criteria are included
- [x] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [x] Dependencies are identified (technical, business, regulatory/policy)
- [x] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by suhafa** [comment]
@dleard As confirmed with Taylor - 
('General Milestone'), - 30 days between substantial deadline and report due date
('Advanced Milestone'), - 30 days between substantial deadline and report due date
('Performance Milestone'), - 150 days. But this is not relevant for milestones any longer as we have Performance milestone covered in the Emissions Intensity screen
('Reporting Milestone') - 30 days between substantial deadline and report due date

As for 'project summary report' type this is something to put on hold, that depends on how we want to display the final milestone type for IA funding stream type. If we have a separate page, we can handle this there. If it stays in milestones, then we can add this in. But in either case, the window is 60 days.

**Comment by matthieu-foucault** [comment]
To reduce future confusion and nomenclature mismatch, could we add a `description` column to our milestone type table?

**Comment by mikevespi** [comment]
@suhafa just to clarify, the date for substantial milestone completion gets auto-populated, but they would still like the ability to change it correct?

**Comment by suhafa** [comment]
@mikevespi absolutely, the substantial milestone completion gets auto-populated on the basis of the report due date.

**Comment by suhafa** [comment]
Update on this ticket: The CIF team clarified during the In-person testing & observation meeting on the 11th of July that they need the **Substantial Completion Deadline** first, and the **Report Due Date** auto-calculated. The report due date is always 30 days after the substantial completion deadline so they don't need the ability to change it. (for performance milestone it is 150 days)

**Comment by Dianadec** [comment]
Reach out to Nan and Suha as needed to clarify older AC.

**Comment by BCerki** [comment]
@suhafa , @nanyangpro , a few questions on this:
- are we populating the due date or the sub completion date? Card title, AC, and comments contradict each other
- the milestone types we have now are Advanced, General, and Reporting--how do those map to the ones listed in the AC?
- could you please provide a link to the updated wireframe mentioned in the comments?

**Comment by nanyangpro** [comment]
Thanks for reaching out, @BCerki! - @suhafa should have more context of this ticket. I'll leave it for her insights.

**Comment by suhafa** [comment]
@BCerki I agree, this ticket needs some updating and streamlining. I'll review this ticket and update you accordingly by today.

**Comment by suhafa** [comment]
@BCerki card title, description, AC and wireframe are all now updated. Hopefully these updates can address the points you've raised. Do review and let me know if this makes sense and is good to go. Thanks Brianna!

**Comment by mikevespi** [comment]
Suha in Teams channel: 
"It would be best to allow that behaviour i.e. every time SCD is changed = RDD reflects the change and vice versa. I say this because RDD is the most important part of the application, not SCD. In case the user wants to pick a particular RDD, I'd rather not have them go calculating SCD to get the right date for RDD if that makes sense"

**Comment by LindsayMacfarlane** [comment]
Thank you @mikevespi ! Tested in dev by editing an existing project and creating a new project. With both, I tried various combinations of entering the substantial due date and report due date. No issues encountered and meets acceptance criteria. Also, huge thank you for updating the acceptance criteria with the updates and making it clear what changed (this really helped me test everything out efficiently :) )
